subsystems/viamserver/viamserver.go

```diff
@@ -348,28 +300,31 @@ func (s *viamServer) Update(ctx context.Context, cfg utils.AgentConfig) (needRes
 	return false
 }
 
-func (s *viamServer) SafeToRestart(ctx context.Context) bool {
+// Property returns a single property of the currently running viamserver.
+func (s *viamServer) Property(ctx context.Context, property string) bool {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	if !s.running || runtime.GOOS == "windows" {
-		return true
-	}
+	switch property {
+	case RestartPropertyRestartAllowed:
+		if !s.running {
+			// Assume agent can restart viamserver if the subsystem is not running.
```

Comment:
We used to also `return true` unconditionally on Windows, meaning a maintenance sensor returning `false` or an active reconfiguration would not stop `viam-agent` from restarting `viam-server` for an update on Windows. That's no longer the case, and both of those things will prohibit `viam-agent` with these changes from restarting `viam-server` (the `WARN` log `viam-server has NOT allowed a restart; will NOT restart` will be output).

Reply:
I've put the hard-coding back for now and filed https://viam.atlassian.net/browse/RSDK-12271 to re-enable at some point in the future. Re-enabling a feature that caused a lot of Windows issues previously as part of this already-large PR is probably too large of a risk.